app/javascript/components/mentoring/discussion/Guidance.tsx

```diff
@@ -45,21 +48,178 @@ export const Guidance = (): JSX.Element => {
+                <a
+                  href="http://ruby-for-beginners.rubymonstas.org/writing_classes/state_and_behaviour.html"
+                  target="_blank"
+                >
+                  This link
+                </a>{' '}
+                may come in handy for a first introduction.
+              </li>
+              <li>
+                <code>unless</code>, inline: With <code>unless</code> instead of{' '}
+                <code>if</code>, we can show what "good" looks like for the
+                conditional statement.
+              </li>
+              <li>
+                <code>error</code>: Custom error message? (Only if the first
+                submission meets the Minimal Solution.)
+              </li>
+              <li>
+                <code>map(&amp;:join)</code>: instead of map with block, but at
+                this point in the track it's okay to just accept it if students
+                use it, no need to require it or dive into the subject of{' '}
+                <code>Symbol#to_proc</code>.
+              </li>
+              <li>
+                <code>each_char</code>: may be preferable over{' '}
+                <code>chars</code> as it returns an <code>Enumerator</code> that
+                will yield each character without creating an intermediate{' '}
+                <code>Array</code>. More: <code>String#chars</code> with a block
+                has a deprecation warning in more recent Ruby versions.{' '}
+                <code>str.chars</code> is a shorthand for{' '}
+                <code>str.each_char.to_a</code>.
+              </li>
+            </ul>
+            <h3>Mentor Research</h3>
+            <ul>
+              <li>
+                The Iteration article mentioned above isn't ideal, but it's one
+                of the few I know of that does more than comparing{' '}
+                <code>each</code> and <code>map</code>, PLUS don't uses hashes
+                for examples. Other suggestions welcome.
+              </li>
+              <li>
+                <code>each_cons</code> raises an ArgumentError for arguments
+                &lt;= 0; it returns the array if the argument &gt;= the array.
+              </li>
+            </ul>
+            {/* TODO: END REPLACE */}
+          </div>
+        </Accordion.Panel>
+      </Accordion>
       <Accordion
         id="solution"
         isOpen={isOpen('solution')}
         onClick={handleClick}
       >
         <Accordion.Header>How you solved the exercise</Accordion.Header>
         <Accordion.Panel>
-          <p>Solution here</p>
-        </Accordion.Panel>
-      </Accordion>
-      <Accordion id="notes" isOpen={isOpen('notes')} onClick={handleClick}>
-        <Accordion.Header>Mentor notes</Accordion.Header>
-        <Accordion.Panel>
-          <p>Notes here</p>
+          {/* TOOD: Replace "#" with mentor_solution.web_url */}
+          <a href="#" className="c-published-solution">
```

Comment:
Note, that there is a normal published-solution component, but has slightly different data from this so I don't think we want to use `Common` component here (at least for now)